src/rdma.c

```diff
@@ -540,6 +540,46 @@ static ssize_t valkeyRdmaRead(valkeyContext *c, char *buf, size_t bufcap) {
     }
 }
 
+static ssize_t valkeyRdmaReadZC(valkeyContext *c, char **buf) {
+    RdmaContext *ctx = c->privctx;
+    long timed, end;
+    uint32_t remained;
+
+    if (valkeyCommandTimeoutMsec(c, &timed)) {
+        return VALKEY_ERR;
+    }
+
+    end = vk_msec_now() + timed;
+
+pollcq:
+    /* try to poll a CQ first */
+    if (connRdmaHandleCq(c) == VALKEY_ERR) {
+        return VALKEY_ERR;
+    }
+
+    if (ctx->recv_offset < ctx->rx_offset) {
+        remained = ctx->rx_offset - ctx->recv_offset;
+        *buf = ctx->recv_buf + ctx->recv_offset;
+        ctx->recv_offset = ctx->rx_offset;
+        return remained;
+    }
+
+    if (valkeyRdmaPollCqCm(c, end) == VALKEY_OK) {
+        goto pollcq;
+    } else {
+        return VALKEY_ERR;
+    }
+}
+
+static void valkeyRdmaReadZCDone(valkeyContext *c) {
+    RdmaContext *ctx = c->privctx;
+    struct rdma_cm_id *cm_id = ctx->cm_id;
+
+    if (ctx->recv_offset == ctx->recv_length) {
+        connRdmaRegisterRx(c, cm_id);
```

Comment:
I don't know the RDMA feature, but `connRdmaRegisterRx` seems to be able to fail when it calls `rdmaSendCommand` ("no empty command buffer" or if `ibv_post_send` fails).

Could there be any problems with the client got stuck/hanging in this case?
Is there a need for a return code in `read_zc_done` to indicate an unrecoverable error?